projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -0,0 +1,93 @@
+import { getRedirectUrl } from '@automattic/jetpack-components';
+import { getSiteFragment } from '@automattic/jetpack-shared-extension-utils';
+import { Modal, Button, CheckboxControl } from '@wordpress/components';
+import { usePrevious } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import './editor.scss';
+
+export const name = 'launchpad-save-modal';
+
+export const settings = {
+	render: function LaunchpadSaveModal() {
+		const isSaving = useSelect(
+			select => select( editorStore ).isSavingNonPostEntityChanges(),
+			[]
+		);
+		const prevIsSaving = usePrevious( isSaving );
+		const [ isModalOpen, setIsModalOpen ] = useState( false );
+		const [ dontShowAgain, setDontShowAgain ] = useState( false );
+		const [ isChecked, setIsChecked ] = useState( false );
+
+		const { launchpadScreenOption, siteIntentOption } = window?.Jetpack_LaunchpadSaveModal || {};
+		const isInsideSiteEditor = window.location.href.includes( 'site-editor' );
```

Comment:
There has to be a better, safer way to check for this. I don't know what it is though. I originally thought to check for an existing post ID, but I'm not sure that's best.

Reply:
The site editor has a wrapper with id="site-editor" and class="edit-site". Beyond that, I'm not so sure what other options we have.

<img width="380" alt="Screen Shot 2022-12-16 at 8 24 42 AM" src="https://user-images.githubusercontent.com/28742426/208108060-9deee4dc-8856-4b6b-9083-ed849dc7af66.png">

Reply:
Turns out we already have a shared utility that can help us here :)

I've switched to it in 8acd02833975f62419f9c98df4deaf660ef0208b

Reply:
Nice, that seems like it should work!

While it works currently, i noted some concerns on that issue as well. And am a bit concerned with that helper in general.

Basically, in the 'early days' of the site editor we used this approach because the edit-site store was never loaded elsewhere. There was some point (I think around the time the template editor entered the post editor maybe) where that approach broke down and the site-editor store was naturally loaded in the post editor, causing breakage due to using this method of determining the editor. It seems the current state is this is not loaded in the post editor, so technically it **should** work as long as that doesnt change again.

In general, that editor determination approach seems fragile due to the above - if any of the stores could ever end up loaded in another editor that might want to leverage their functionality. And currently the order in that utility is highly important as `core/edit-post` is loaded in other editors such as the site editor.

All things considered, I would think that the site editor id might be the most stable.

Reply:
Thanks for the extra info!

> All things considered, I would think that the site editor id might be the most stable.

That works for me. I've just pushed c039ce590ac4cd56c41f243fc2f04410eeed44e4